### DISCUSSION TITLE
Perception of OGS Activity

### DISCUSSION THREAD

**Discussion by kevincoleman** [online-go/online-go.com#2370] [Ideas]
This is obviously a potentially huge topic, but one of the most obvious metrics of how OGS is doing is just the general amount of game traffic. Thought I’d start a discussion here so we can explore and talk about some ideas around the traffic and how to increase it (assuming we do want to).

When users hit the site and go to play a game they get a visual of how active the site is, right here. I’m aware this is a visualization of games listed, not games being played, but it’s still an indicator:
![Screenshot 2023-09-18 at 7 54 48 AM](https://github.com/online-go/online-go.com/assets/445707/f565f1b8-b392-4492-b3a7-0ca9f6a6ff74)
Interestingly, this displays vastly differently just depending on their filtering settings. I don’t propose we change that, but it’s worth noting that if a user just kinda forgets they have any filters set they’ll get a different view of how active the site is from someone else.

This screen basically has the power to engage or disengage someone. The line of reasoning:

## Case 1 (under trafficked)
“Doesn’t look like there’s much going on”

### Options:
1. Go through the effort of creating and listing some games
2. Wait until they see a game they like, then try to snatch it (how long people wait would be a useful metric, if we can record that)
3. Change filters to maybe find more games outside what they saw when they got the above impression
4. Do other things on the site
5. Leave

## Case 2 (the “happy path”)
“Nice, plenty of games to choose from”

### Options:
1. Pick a game to add and start playing
2. List their own game because they have a certain idea in mind
3. Do other things on the site.
4. Leave

### Options:
1. Try to find a game in the long list, or using the crowded chart
2. Use filters smartly to filter the list/chart to find a game
3. Do other things on the site
4. Leave

I think it’s fair to say we probably hover between Case 1 and Case 2 for most players, depending on how picky someone is. The Quick Match feature helps people bypass this flow if they’re less picky about what games they play. Each of these cases probably has a different conversion rate of getting someone through to play a game. Do we gather detailed stats on the user flows through this page? Seems like a key place to start.

I suspect we’d want to hover more between case 2 and case 3. Even if people have a slightly harder time finding a game (as in case 3) it’s the “better” problem to have as users get the perception that this place is really hoppin’. Then we can just optimize toward helping people use the interface smartly to get the right game for them. If this is true, then we want to find ways to help people list more games, more often. I’ve got ideas for how to make this happen, but before I go nuts on those I wanted to post here and validate/refute any of these assumptions. If this seems right maybe we can make this discussion a jump-off point for linking to ideas for increasing game traffic.

Open to thoughts on this! Am I way off? What are the insights I’m missing?

**Comment by anoek** [comment]
> Do we gather detailed stats on the user flows through this page? Seems like a key place to start.

We don't do that anymore, there has been some notable pushback and concerns about using things like google analytics in the past so we've stopped using them.

However yes, there was a lot of flow back and forth between play and home, and from those pages to the Game page, then back again. 

> If this is true, then we want to find ways to help people list more games, more often. 

I don't think optimizing on creating custom games (which is what the list shows) is necessarily what we want as it's fundamentally not a scalable matchmaking system, automatch better serves that I think. Custom games certainly serve their purpose though, allowing folks to find matches with non standard settings, so it's still good to keep around I think.

If the goal is to show "activity" of sorts, a couple of other potential things come to mind. 

 * Show auto match state potentially (like who is in the queue, or at least their ranks), 
 * Incorporate the connected players and live game count somewhere
 * Show a mini goban of the highest ranked fast paced game on the site

Lichess also incorporates things like listing open tournaments and a "Puzzle of the day" system. I've long wanted to do a puzzle of the day kind of thing, might be something of interest on on a landing page of sorts perhaps.

**Reply by kevincoleman** [reply]
This all makes good sense. I myself don't love trackers. In this case we'd be using the key data to inform our developmental decisions, so I thought it was worth asking. I wonder—if we collected our own fully anonymized stats on key things we're considering changing would stir the pot too much? The alternative is us making decisions without data, which is kind of like trying to guess your way across town blindfolded.

I don't think we really need to expedite the custom game creation mechanic either. That'll always be a gauntlet of a form with great granular controls. I'm thinking more that we could feature the game creation UI (or a simplified version) at the end of a game, on the empty games page (as we've been discussing already), or during a new user startup intro. Making it extra clear in the nav bar may help people find the creation screen better too.

I like your ideas around featuring the existing activity more. There may also be potential here re: getting people to come just to watch games.

All this said, I'm operating under the assumption that we want to increase the site's engagement and traffic from where it is. Is that a decent assumption?

**Reply by anoek** [reply]
> if we collected our own fully anonymized stats on key things we're considering changing would stir the pot too much?

Sure, I think it's more of a question of figuring out what questions we want to answer and then doing whatever work required to wire up those stats. Some things, like how many games are being played, how many puzzles solved, stuff like that should be fairly easy to do. Other things, like tracking flows between pages and whatnot would require more work obviously.


> I'm operating under the assumption that we want to increase the site's engagement and traffic from where it is. Is that a decent assumption?

Sure, generally speaking those things mean more people are having more fun on the site, sounds great :)

As for growth it's probably worth noting that at this point most of the established western Go community is aware of the site and a good fraction, I'd wager the majority, at least have an account here. That means growth comes from:

  1) Brand new western players who are just learning about the game
  2) Enticing existing players that currently still favor other servers to come play here more
  3) Figuring out how to better reach the Eastern player base and entice them to visit more

My general feeling is that for (1) we need to capture their interest and guide them through learning the game, either through lessons and/or playing against weak bots, (2) is largely a still an artifact of other servers being able to find good matches quicker for some rank bands, and/or they simply don't like the way OGS works, and (3) is lack of exposure, lack of compelling reasons to change from the services they know, and likely in some cases find the match making slower than they'd like compared to the larger eastern servers.
